src/app/app.component.ts

```diff
@@ -1,10 +1,50 @@
 import { Component } from '@angular/core';
 
+class Point {
+  x: number;
+  y: number;
+
+  constructor(xNum: number, yNum: number) {
+    this.x = xNum;
+    this.y = yNum;
+  }
+  add(newPoint: Point){
+    return newPoint = new Point(this.x, + this.y);
+  }
+}
+
+class Point3D extends Point{
+  // x: number;
+  // y: number;
+  z: number;
+
+  constructor(xNum, yNum, zNum){
+    super(Point, Point3D);
+    this.x = xNum;
+    this.y = yNum;
+    this.z = zNum;
+    add(new3DPoint: Point3D){
+      return new3DPoint = new Point3D(xNum, + yNum, + zNum);
```

Comment:
this will create a new 3D Point with existing 3D Point's values. you need to add the two Points here